linera-storage/src/db_storage.rs

```diff
@@ -1157,3 +1163,168 @@ where
+        }
+        BaseKey::Blob(blob_id) => {
+            let root_key = RootKey::Blob(blob_id).bytes();
+            Ok((root_key, DEFAULT_KEY.to_vec()))
+        }
+        BaseKey::BlobState(blob_id) => {
+            let root_key = RootKey::Blob(blob_id).bytes();
+            Ok((root_key, ONE_KEY.to_vec()))
+        }
+        BaseKey::Event(event_id) => {
+            let root_key = RootKey::Event(event_id.chain_id).bytes();
+            let key = event_key(&event_id);
+            Ok((root_key, key))
+        }
+        BaseKey::BlockExporterState(index) => {
+            let root_key = RootKey::BlockExporterState(index).bytes();
+            Ok((root_key, EMPTY_KEY.to_vec()))
+        }
+        BaseKey::NetworkDescription => {
+            let root_key = RootKey::NetworkDescription.bytes();
+            Ok((root_key, DEFAULT_KEY.to_vec()))
+        }
+    }
+}
+
+impl<Database, C> DbStorage<Database, C>
+where
+    Database: KeyValueDatabase + Clone + Send + Sync + 'static,
+    Database::Store: KeyValueStore + Clone + Send + Sync + 'static,
+    C: Clock + Clone + Send + Sync + 'static,
+    Database::Error: Send + Sync,
+{
+    async fn migrate_key_set(
+        &self,
+        first_byte: &u8,
+        base_keys: Vec<Vec<u8>>,
+    ) -> Result<(), ViewError> {
+        tracing::info!(
+            "migrate_key_set with first_byte={first_byte} for |base_keys|={}",
+            base_keys.len()
+        );
+        for (index, chunk_base_keys) in base_keys.chunks(BLOCK_KEY_SIZE).enumerate() {
+            tracing::info!(
+                "index={index} processing chunk of size {}",
+                chunk_base_keys.len()
+            );
+            let store = self.database.open_shared(&[])?;
+            let values = store
+                .read_multi_values_bytes(chunk_base_keys.to_vec())
+                .await?;
+            let values = values
+                .into_iter()
+                .map(|value| value.ok_or(ViewError::MissingEntries))
+                .collect::<Result<Vec<Vec<u8>>, ViewError>>()?;
+            let mut batch = MultiPartitionBatch::new();
+            for (base_key, value) in chunk_base_keys.iter().zip(values) {
+                tracing::info!("base_key={base_key:?} value={value:?}");
+                let (root_key, key) = map_base_key(base_key)?;
+                batch.put_key_value_bytes(root_key, key, value);
+            }
+            self.write_batch(batch).await?;
+            // Now delete the keys
+            let mut batch = Batch::new();
+            for key in chunk_base_keys {
+                batch.delete_key(key.to_vec());
+            }
+            store.write_batch(batch).await?;
+        }
+        Ok(())
+    }
+
+    async fn migrate_0_to_1(&self) -> Result<(), ViewError> {
+        for first_byte in MOVABLE_KEYS_0_1 {
+            let store = self.database.open_shared(&[])?;
+            let keys = store.find_keys_by_prefix(&[*first_byte]).await?;
```

Comment:
Could this be an issue for Rocksdb on testnet without https://github.com/linera-io/linera-protocol/pull/4859 ?